File: ts/earthbucks-lib/test/iso-hex.test.ts
import { describe, expect, test } from "vitest";
import { IsoHex } from "../src/iso-hex";
import { EbxBuffer } from "../src/ebx-buffer";

describe("strictHex", () => {
  test("should return true for valid hex strings", () => {
    expect(IsoHex.isValid("00")).toBe(true);
    expect(IsoHex.isValid("1234567890abcdef")).toBe(true);
    expect(IsoHex.isValid("1234567890abcde")).toBe(false);
  });

  test("should return false for invalid hex strings", () => {
    expect(IsoHex.isValid("0")).toBe(false);
    expect(IsoHex.isValid("0g")).toBe(false);
    expect(IsoHex.isValid("1234567890abcdeF")).toBe(false);
  });

  test("encode/decode", () => {
    const buffer = EbxBuffer.from("1234567890abcdef", "hex");
    const hex = IsoHex.encode(buffer);
    const decodedEbxBuffer = IsoHex.decode(hex).unwrap();
    expect(decodedEbxBuffer).toEqual(buffer);
  });
});

File: rs/earthbucks_lib/tests/iso_hex_test.rs
use earthbucks_lib::iso_hex::*;

#[test]
fn test_is_valid() {
    assert!(is_valid("00"));
    assert!(is_valid("1234567890abcdef"));
    assert!(!is_valid("1234567890abcde"));
    assert!(!is_valid("0"));
    assert!(!is_valid("0g"));
    assert!(!is_valid("1234567890abcdeF"));
}

#[test]
fn test_encode_decode() {
    let buffer = hex::decode("1234567890abcdef").unwrap();
    let hex = encode(&buffer);
    let decoded_buffer = decode(&hex).unwrap();
    assert_eq!(decoded_buffer, buffer);
}
